packit_service/worker/handlers/open_scan_hub.py

```diff
@@ -99,17 +100,71 @@ class OpenScanHubTaskFinishedHandler(
     event: OpenScanHubTaskFinishedEvent
     task_name = TaskName.openscanhub_task_finished
 
+    @staticmethod
+    def check_new_findings_identified(url: str) -> bool:
+        """
+        Downloads a JSON file from the specified URL and
+        checks if the 'defects' array is empty.
+
+        Parameters:
+            url (str): The URL of the JSON file.
+
+        Returns:
+            bool: True if the 'defects' array is not empty or we were not
+            able to get it, False otherwise.
+        """
+        logger.info(
+            f"About to check if there were any new findings identified by the scan in {url}."
+        )
+        try:
+            with requests.get(url, timeout=10) as response:
+                response.raise_for_status()
+                data = response.json()
+
+                defects = data.get("defects", [])
+                return len(defects) > 0
+
+        except requests.exceptions.RequestException:
+            logger.error("Error occurred while trying to download the JSON file.")
+            return True
+        except json.JSONDecodeError:
+            logger.error("The response is not a valid JSON.")
+            return True
+
+    @staticmethod
+    def get_issues_added_url(
+        task_id: int,
+        openscanhub_url: str = "https://openscanhub.fedoraproject.org",
+        file_format: str = "html",
+    ) -> str:
+        """
+        Constructs the URL for the added issues in the specified
+        format for the given OpenScanHub task.
+
+        Parameters:
+            task_id (int): The ID of the task.
+            openscanhub_url (str)
+            file_format (str): The format of the added issues file ('html' or 'json').
+
+        Returns:
+            str: The full URL to access the added issues in the specified format.
+        """
+
+        return f"{openscanhub_url}/task/{task_id}/log/added.{file_format}"
+
     def run(self) -> TaskResults:
         self.check_scan_and_build()
 
         if self.event.status == OpenScanHubTaskFinishedEvent.Status.success:
             state = BaseCommitStatus.success
-            description = "Scan in OpenScanHub is finished. Check the URL for more details."
-            external_links = {
-                "Added issues": self.event.issues_added_url,
-                "Fixed issues": self.event.issues_fixed_url,
-                "Scan results": self.event.scan_results_url,
-            }
+
+            if self.check_new_findings_identified(self.event.issues_added_url):
+                description = "Scan in OpenScanHub is finished. Check the URL for more details."
```

Comment:
My concern is that at this point you are treating a couple of exceptions and newly identified findings the same way. This may not be intuitive for an average user and whether it is a newly identified finding or an exception should be kept transparent to the user.